src/main/kotlin/org/rust/ide/console/RsConsoleRunner.kt

```diff
@@ -0,0 +1,226 @@
+
+    override fun fillToolBarActions(
+        toolbarActions: DefaultActionGroup,
+        defaultExecutor: Executor,
+        contentDescriptor: RunContentDescriptor
+    ): MutableList<AnAction> {
+        val actionList = arrayListOf<AnAction>(
+            createConsoleExecAction(consoleExecuteActionHandler),
+            RestartAction(this)
+        )
+        toolbarActions.addAll(actionList)
+        return actionList
+    }
+
+    fun runSync(requestEditorFocus: Boolean) {
+        if (checkNeedInstallEvcxr()) return
+
+        try {
+            initAndRun()
+            ProgressManager.getInstance().run(object : Task.Backgroundable(project, "Connecting to Console", false) {
+                override fun run(indicator: ProgressIndicator) {
+                    indicator.text = "Connecting to console..."
+                    connect()
+                    if (requestEditorFocus) {
+                        consoleView?.requestFocus()
+                    }
+                }
+            })
+        } catch (e: ExecutionException) {
+            LOG.warn("Error running console", e)
+            showErrorsInConsole(e)
+        }
+    }
+
+    fun run(requestEditorFocus: Boolean) {
+        if (checkNeedInstallEvcxr()) return
+        TransactionGuard.submitTransaction(project, Runnable { FileDocumentManager.getInstance().saveAllDocuments() })
+
+        ApplicationManager.getApplication().executeOnPooledThread {
+            ProgressManager.getInstance().run(object : Task.Backgroundable(project, "Connecting to Console", false) {
+                override fun run(indicator: ProgressIndicator) {
+                    indicator.text = "Connecting to console..."
+                    try {
+                        initAndRun()
+                        connect()
+                        if (requestEditorFocus) {
+                            consoleView?.requestFocus()
+                        }
+                    } catch (e: Exception) {
+                        LOG.warn("Error running console", e)
+                        UIUtil.invokeAndWaitIfNeeded(Runnable { showErrorsInConsole(e) })
+                    }
+                }
+            })
+        }
+    }
+
+    private fun checkNeedInstallEvcxr(): Boolean {
+        val needInstallEvcxr = Cargo.checkNeedInstallEvcxr(project)
+        if (needInstallEvcxr) {
+            ApplicationManager.getApplication().invokeLater {
+                RsConsoleToolWindow.getInstance(project).hide()
+            }
+        }
+        return needInstallEvcxr
+    }
+
+    override fun initAndRun() {
+        UIUtil.invokeAndWaitIfNeeded(Runnable {
+            super.initAndRun()
+
+            consoleExecuteActionHandler.sendText(":opt 0\n")
+        })
+    }
+
+    override fun createProcessHandler(process: Process): OSProcessHandler =
+        RsConsoleProcessHandler(
+            process,
+            consoleView,
+            consoleCommunication,
+            commandLine!!.commandLineString,
```

Comment:
Could `commandLine` be `null` here? If so, consider using something like `commandLine?.commandLineString.orEmpty()` instead